hypothesis-python/src/hypothesis/strategies/_internal/strategies.py

```diff
@@ -585,7 +595,9 @@ def do_filtered_draw(self, data):
         # Before building the list of allowed indices, speculatively choose
         # one of them. We don't yet know how many allowed indices there will be,
         # so this choice might be out-of-bounds, but that's OK.
-        speculative_index = data.draw_integer(0, max_good_indices - 1)
+        speculative_index = data.draw_integer(
+            0, max_good_indices - 1, shrink_towards=self.shrink_towards
+        )
```

Comment:
Hmm... having gone to all this effort to plumb through `shrink_towards`; I wonder if we actually just want a `reverse_elems` boolean so that our speculation will efficiently work from the end that we want to shrink to.

Or.  Having said that, we could just `self.elements = bundle[::-1]` in `Bundle.do_draw` 🤦‍♂️